src/model/SVGFile.java

```diff
@@ -34,15 +43,34 @@ public void setXmlcontent(String xmlcontent) {
 		this.xmlcontent = xmlcontent;
 	}
 
+	
+	/*
+	 * save SVG file into files
+	 */
 	@Override
-	public boolean writeSVGFile(String content) {
-		// TODO Auto-generated method stub
-		return false;
+	public boolean writeSVGFile() {
+		file = new File(String.format(filnename, this.getXmlcontent()));
+		if(file.exists()){
+			System.out.println("WARNING : the file "+this.getFilnename()+" already exists!");
+			return false;
+		}else{
+			try {
+				out = new BufferedWriter(new FileWriter(file));
+				out.write(this.getXmlcontent());
+				out.close();
+				return true;
+			} catch (IOException e) {
+				e.printStackTrace();
+				return false;
+			} 
+		}
 	}
 
 	@Override
-	public boolean addShapes(List<Shapes> listShapes) {
-		// TODO Auto-generated method stub
+	public boolean addShapes(List<ShapesType> listShapes) {
```

Comment:
Not sure that function is here. There is already a function for that. And how do you know where to add the shapes?